radix-engine/src/track/track.rs

```diff
@@ -903,18 +960,32 @@ impl<'s, S: SubstateDatabase, M: DatabaseKeyMapper + 'static> SubstateStore for
             new_updates
         };
 
+        let num_iterations = tracked_iter.num_iterations;
+        drop(tracked_iter);
```

Comment:
Minor: probably better to use standard use of `{}` to limit scoping and adding `tracked_iter.num_iterations;` into the above code block